src/cycax/cycad/Part.py

```diff
@@ -0,0 +1,282 @@
+        side: str,
+        part_no: str,
+        x_size: float,
+        y_size: float,
+        z_size: float,
+    ):
+        super().__init__(x, y, z, side)
+        self.part_no = part_no
+        self.x_size = x_size
+        self.y_size = y_size
+        self.z_size = z_size
+        self.features = []  # Stores all the holes to be cut
+        self.move_holes = []
+        self.x_max: float = self.x_size  # Right
+        self.x_min: float = 0  # Left
+        self.y_min: float = 0  # Front
+        self.y_max: float = self.y_size  # Back
+        self.z_min: float = 0  # Bottom
+        self.z_max: float = self.z_size  # Top
+        self.bounding_box = {}
+        self.rotmax = [self.x_size, self.y_size, self.z_size]
+        self.moves = [0, 0, 0]
+        self.rotate = [0, 0, 0]
+        self.final_location = False
+
+    def make_hole(
+        self,
+        x: float,
+        y: float,
+        side: str,
+        diameter: float,
+        depth: float,
+        Inner: bool = True,
+    ):
+        """!!!!!THIS METHOD WILL ONLY WORK IF WE ARE MAKING HOLES IN THE CENTRE OF A CUBIC OBJECT, NEED TO RETHINK LOGIC!!!!!!
+        If instead of top and bottom it were possible to think rather (x, y, z_max) it would be more affective as the guessing wouldn't need to occur. These if statements wouldn't and is would work even when not a square.
+        It would however require more lines of code to be written on the user end."""
+        if side == TOP:
+            temp_hole = Holes(side, x, y, self.z_max, diameter, depth)
+        if side == BOTTOM:
+            temp_hole = Holes(side, x, y, self.z_min, diameter, depth)
+        if side == LEFT:
+            temp_hole = Holes(side, self.x_min, x, y, diameter, depth)
+        if side == RIGHT:
+            temp_hole = Holes(side, self.x_max, x, y, diameter, depth)
+        if side == FRONT:
+            temp_hole = Holes(side, x, self.y_min, y, diameter, depth)
+        if side == BACK:
+            temp_hole = Holes(side, x, self.y_max, y, diameter, depth)
+        else:
+            exit
+        # This will add it to the relevnt array
+        if Inner:
+            self.features.append(temp_hole)
+        else:
+            self.features.append(temp_hole)
+            self.move_holes.append(temp_hole)
+
+    def make_slot(
+        self,
+        x: float,
+        y: float,
+        side: str,
+        x_size: float,
+        y_size: float,
+        z_size: float,
+        horizontal: bool = True,
+        Inner=True,
+    ):
+        """!!!!!THIS METHOD WILL ONLY WORK IF WE ARE MAKING HOLES IN THE CENTRE OF A CUBIC OBJECT, NEED TO RETHINK LOGIC!!!!!!
+        If instead of top and bottom it were possible to think rather (x, y, z_max) it would be more affective as the guessing wouldn't need to occur. These if statements wouldn't and is would work even when not a square.
+        It would however require more lines of code to be written on the user end."""
+        if side == TOP:
```

Comment:
Consider this long bunch of if statements and thing a year in to the future. You'll look at this and go "whaaaaat!". At a quick glance it is hard to see why the if is there.

First consider using an if/elif/elif/else code block.
Only call Slot() once.
Process variables before calling Slot(), if you (or me) want to add a debug to log line in here it has to go in a few places.
There are many ways to do that. 
An if for each side and set x,y, and z. 
Say:
```
if side == 'TOP':
   x=x
   y=y
   z=self.z_max
elif side == "BOTTOM":
   x=x
   y=y
   z=self.z_max
```

Or per variable:
```
x=x
if side == 'LEFT'
    x = self.x_max
if side == 'RIGHT'
    x == self.x_min
```

FOR FUN! Dont use this, its an FYI.
Python has no case/switch statement like you would use in Java. But a common pattern is to abuse dictionaries.
```
x,y,z = {'TOP': (x,y,self.z_max),
             "BOTTOM": (x,y,self.z_min),
            ...
            }.get(side)
```
Its very clean and reads easy, only manipulates data and short so fits into my brain easily.